src/v/archival/ntp_archiver_service.cc

```diff
@@ -780,6 +781,16 @@ ss::future<> ntp_archiver::maybe_flush_manifest_clean_offset() {
 
 ss::future<cloud_storage::upload_result> ntp_archiver::upload_manifest(
   std::optional<std::reference_wrapper<retry_chain_node>> source_rtc) {
+    if (!_feature_table.local().is_active(
+          features::feature::cloud_storage_manifest_format_v2)) {
+        vlog(
+          archival_log.info,
+          "Skipping manifest upload until all nodes in the cluster have been "
+          "upgraded.");
+
+        co_return cloud_storage::upload_result::cancelled;
```

Comment:
Is the intent here that we won't be able to serialize v1 manifests at all after the new manifest compression lands, so we don't bother?

Otherwise, if intend on keeping the ability to serialize v1, we could do that here, no?

Reply:
> Is the intent here that we won't be able to serialize v1 manifests at all after the new manifest compression lands, so we don't bother?

In theory, we could serialise a v1 manifest, although the interfaces for this don't exist at the moment. This check guards against the nodes in the cluster that don't know how to parse v2 manifests receiving one. It's a bit of an edge case as that can only happen during recovery.

> Otherwise, if intend on keeping the ability to serialize v1, we could do that here, no?
Yes, I don't know if it's worth the extra complexity though.

Reply:
Makes sense, given we will need to do something similar for the upcoming manifest format / segment name changes anyway